ruint-macro/src/lib.rs

```diff
@@ -162,15 +189,25 @@ fn transform_tree(tree: TokenTree) -> TokenTree {
 }
 
 /// Iterate over a [`TokenStream`] and transform all [`TokenTree`]s.
-fn transform_stream(stream: TokenStream) -> TokenStream {
-    stream.into_iter().map(transform_tree).collect()
+fn transform_stream(stream: TokenStream, base_type: LiteralBaseType) -> TokenStream {
+    stream
+        .into_iter()
+        .map(|tree| transform_tree(tree, base_type))
+        .collect()
 }
 
 // Repeat the crate doc
 #[doc = include_str!("../Readme.md")]
 #[proc_macro]
 pub fn uint(stream: TokenStream) -> TokenStream {
-    transform_stream(stream)
+    transform_stream(stream, LiteralBaseType::Uint)
+}
+
+// Repeat the crate doc
+#[doc = include_str!("../Readme.md")]
+#[proc_macro]
+pub fn bits(stream: TokenStream) -> TokenStream {
```

Comment:
One of the nice things about the `uint!` macro is that you can do this:

```
uint!{

let a = 42_U256;
let b = 0xf00f_1337_c0d3_U256;
let c = a + b;
assert_eq!(c, 263947537596669_U256);

}
```

I'd like to keep that behaviour and extend it with support for `B` so you can mix both:

```
uint!{

let foo = 0xf0f1337_U32;
let goo = 0x90bb1e_B32;

}
```